webpack/assets/javascripts/react_app/components/SearchBar/SearchBar.js

```diff
@@ -17,15 +20,16 @@ const SearchBar = ({
     <div className="search-bar input-group">
       <AutoComplete
         id={autocomplete.id}
-        handleSearch={() => resolveSearchQuery(searchQuery)}
-        initialQuery={autocomplete.searchQuery || ''}
+        handleSearch={() => resolveSearchQuery(searchQuery, onSearch)}
+        initialQuery={initialQuery || autocomplete.searchQuery || ''}
```

Comment:
can you reuse the `autocomplete` object and pass the `initialQuery` as `searchQuery` instead ?

Reply:
we should not mix data from the server with data from the client, it will be very confusing. it is an initialQuery, and it will only be updated on back/forward navigation using react-router.